File: com.g2.backend.tiendaropa/src/main/java/com/g2/backend/tiendaropa/com/model/entity/Producto.java
package com.g2.backend.tiendaropa.com.model.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Producto {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String nombre;
    private String descripcion;
    private String categoria;
    private Double precio;
    private Integer stock;
    private String image;

    // Relación con Carrito
    @ManyToMany(mappedBy = "productos") // Cambiado a "productos" para coincidir con la entidad Carrito
    private List<Carrito> carritos = new ArrayList<>();
}

File: src/app/producto.ts
export class Producto {
  id: number;
  nombre: string;
  descripcion: string;
  categoria: string;
  precio: number;
  stock: number;
  image: string;

  constructor(id: number, nombre: string, descripcion: string, categoria: string, precio: number, stock: number, image: string) {
    this.id = id;
    this.nombre = nombre;
    this.descripcion = descripcion;
    this.categoria = categoria;
    this.precio = precio;
    this.stock = stock;
    this.image = image;
  }
}